services/harmony/app/frontends/service-image-tags.ts

```diff
@@ -130,29 +130,32 @@ async function validateTaggedImageIsReachable(
 }
 
 /**
- * Returns value of the enabled field of service_deployment table.
+ * Returns the values of the enabled and message fields of the service_deployment table.
  * @param tx - The database transaction
- * @returns The boolean value of the enabled field
+ * @returns An object containing the boolean value of the enabled field and the message string
  */
-async function getEnabled(): Promise<boolean> {
+async function getEnabledAndMessage(): Promise<{ enabled: boolean, message: string }> {
   let enabled = true;
+  let message = '';
   let results = null;
   await db.transaction(async (tx) => {
-    results = await tx('service_deployment').select('enabled');
+    results = await tx('service_deployment').select('enabled', 'message');
   });
 
   if (results[0].enabled === 0 || results[0].enabled === false) {
     enabled = false;
   }
-  return enabled;
+  [{ message }] = results;
+
+  return { enabled, message };
 }
 
 /**
  * Set the enabled to true in service_deployment table
  */
-export async function enableServiceDeployment(): Promise<void> {
+export async function enableServiceDeployment(message: string): Promise<void> {
   await db.transaction(async (tx) => {
-    await tx('service_deployment').update({ enabled: true });
+    await tx('service_deployment').update({ enabled: true, message: message });
```

Comment:
We need to truncate to less than 4096 characters (we have a truncate function we use for other database fields with a fixed length). Otherwise we return a 500 error to the end user with an HTML page.

I think there's something in `error.ts` to make sure we return JSON for some endpoints instead of HTML - might need to add the service-deployments-state route to that.